pkg/api/types.go

```diff
@@ -301,24 +301,43 @@ type DcosConfig struct {
+	SourcePortRanges           []string `json:"sourcePortRanges"`
+	DestinationAddressPrefix   string   `json:"destinationAddressPrefix"`
+	DestinationAddressPrefixes []string `json:"destinationAddressPrefixes"`
+	DestinationPortRange       string   `json:"destinationPortRange"`
+	DestinationPortRanges      []string `json:"destinationPortRanges"`
+}
+
+// MasterNetworkAccessProfile describes the inbound network configuration for access the master nodes.
+type MasterNetworkAccessProfile struct {
+	TLS *NetworkAccessProfile `json:"tls,omitempty"`
+	SSH *NetworkAccessProfile `json:"ssh,omitempty"`
+}
+
 // MasterProfile represents the definition of the master cluster
 type MasterProfile struct {
-	Count                    int               `json:"count"`
-	DNSPrefix                string            `json:"dnsPrefix"`
-	VMSize                   string            `json:"vmSize"`
-	OSDiskSizeGB             int               `json:"osDiskSizeGB,omitempty"`
-	VnetSubnetID             string            `json:"vnetSubnetID,omitempty"`
-	VnetCidr                 string            `json:"vnetCidr,omitempty"`
-	FirstConsecutiveStaticIP string            `json:"firstConsecutiveStaticIP,omitempty"`
-	Subnet                   string            `json:"subnet"`
-	IPAddressCount           int               `json:"ipAddressCount,omitempty"`
-	StorageProfile           string            `json:"storageProfile,omitempty"`
-	HTTPSourceAddressPrefix  string            `json:"HTTPSourceAddressPrefix,omitempty"`
-	OAuthEnabled             bool              `json:"oauthEnabled"`
-	PreprovisionExtension    *Extension        `json:"preProvisionExtension"`
-	Extensions               []Extension       `json:"extensions"`
-	Distro                   Distro            `json:"distro,omitempty"`
-	KubernetesConfig         *KubernetesConfig `json:"kubernetesConfig,omitempty"`
+	Count                    int                         `json:"count"`
+	DNSPrefix                string                      `json:"dnsPrefix"`
+	VMSize                   string                      `json:"vmSize"`
+	OSDiskSizeGB             int                         `json:"osDiskSizeGB,omitempty"`
+	VnetSubnetID             string                      `json:"vnetSubnetID,omitempty"`
+	VnetCidr                 string                      `json:"vnetCidr,omitempty"`
+	FirstConsecutiveStaticIP string                      `json:"firstConsecutiveStaticIP,omitempty"`
+	Subnet                   string                      `json:"subnet"`
+	IPAddressCount           int                         `json:"ipAddressCount,omitempty"`
+	StorageProfile           string                      `json:"storageProfile,omitempty"`
+	HTTPSourceAddressPrefix  string                      `json:"HTTPSourceAddressPrefix,omitempty"`
+	OAuthEnabled             bool                        `json:"oauthEnabled"`
+	NetworkAccessProfile     *MasterNetworkAccessProfile `json:"networkAccessProfile,omitEmpty"`
```

Comment:
I'm curious why you chose to build a struct `MasterNetworkAccessProfile` with hardcoded `SSH` and `TLS` properties instead of builds an array of `NetworkAccessProfile`s, i.e: 
```go
NetworkAccessProfiles []NetworkAccessProfile `json:"networkAccessProfiles,omitEmpty"
```

Wouldn't that be more flexible if we want to add more rules to the NSG in the future ?

Reply:
Definitely. I kept the scope of the change limited to a precise purpose to facilitate the discussion a little. With that, I'm obviously happy to adjust once we agree on the general approach.

Reply:
I agree with @sylr that we want to ship this with a general implementation, as changing the type spec later will be costly.

Reply:
ACK, how would we map to the individual nsg's then? Right now, when applying the template, I do an explicit selection. Maybe a map indexed with the name is more flexible and still allows us to select the right profile for a specific NSG.